lib/ansible/modules/system/filesystem.py

```diff
@@ -107,21 +108,18 @@ def _get_fs_size(fssize_cmd, dev, module):
                     block_count = int(col[3].split(',')[0])
                     break
         else:
-            module.fail_json(msg="Failed to get block count and block size of %s with %s" % (dev, cmd), rc=rc, err=err )
-    elif 'btrfs' == fssize_cmd:
-        #ToDo
-        # There is no way to get the blocksize and blockcount for btrfs filesystems
-        block_size = 1
-        block_count = 1
+            module.fail_json(msg="Failed to get block count and block size of %s with %s" % (dev, cmd), rc=rc, err=err)
     elif 'pvs' == fssize_cmd:
         rc, size, err = module.run_command([cmd, '--noheadings', '-o', 'pv_size', '--units', 'b', dev])
         if rc == 0:
             block_count = int(size[:-1])
             block_size = 1
         else:
-            module.fail_json(msg="Failed to get block count and block size of %s with %s" % (dev, cmd), rc=rc, err=err )
+            module.fail_json(msg="Failed to get block count and block size of %s with %s" % (dev, cmd), rc=rc, err=err)
+    else:
+        module.fail_json(changed=False, msg="module does not support resizing %s filesystem yet." % module.param['fstype'])
```

Comment:
This can be considered a bit of nitpicking perhaps, and other reviewers may think it does not matter much, but - wouldn't it be better to move this particular failure condition + message outside of this function and somewhere into the `main()` as part of verifying the parameters?

Reply:
> Would it be hard to implement proper support for it? I have not used btrfs myself, so no idea if this would be very complicated or not?

It appears that `btrfs filesystem resize /mounted/path`. I propose that I implement that in another PR.

> wouldn't it be better to move this particular failure condition + message outside of this function and somewhere into the main() as part of verifying the parameters?
This is now tested into the `main` method.